x/staking/keeper/delegation.go

```diff
@@ -643,9 +643,22 @@ func (k Keeper) Delegate(
 	// validator.Tokens
 	lastTotalPower := k.GetLastTotalPower(ctx)
 	maxVotingPowerEnforcementThreshold := k.MaxVotingPowerEnforcementThreshold(ctx)
-	if lastTotalPower.GTE(maxVotingPowerEnforcementThreshold) {
-		newVotingPowerRatio := sdk.NewDecFromInt(validator.Tokens.Add(bondAmt)).Quo(sdk.NewDecFromInt(lastTotalPower))
-		if newVotingPowerRatio.GT(k.MaxVotingPowerRatio(ctx)) {
+
+	// 1 power = Bond Amount / Power Reduction
+	validatorAddtionalPower := bondAmt.Quo(k.PowerReduction(ctx))
+	newTotalPower := lastTotalPower.Add(validatorAddtionalPower)
+
+	// If it's beyond genesis then enforce power ratio per validator if there's more than maxVotingPowerEnforcementThreshold
+	if newTotalPower.GTE(maxVotingPowerEnforcementThreshold) && ctx.BlockHeight() > 0 {
```

Comment:
is the intended behavior here that voting power caps will only be enforced AFTER the max voting power exceeds the threshold, and prior to that it can be as imbalanced as it wants to be?

Reply:
Based on the original design that does seem to be the case, the idea is that the power ratio will only be enforced after the chain exceeds a certain total power threshold. 

cc @LCyson

Reply:
this was for allowing us to have flexibility to decide when we want to enforce this voting max cap (esp for the first few blocks if we want to delegate majority voting power to our own nodes for example). But for mainnet the best practice is prob to start enforcing the max power after the genesis block